source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -543,6 +578,71 @@ void CacheFilter::finalizeEncodingCachedResponse() {
   filter_state_ = FilterState::ResponseServedFromCache;
 }
 
+LookupStatus CacheFilter::lookupStatus() const {
+  if (lookup_result_ == nullptr && lookup_ != nullptr) {
+    return LookupStatus::RequestIncomplete;
+  }
+
+  if (lookup_result_ != nullptr) {
+    switch (lookup_result_->cache_entry_status_) {
+    case CacheEntryStatus::Ok:
+      return LookupStatus::CacheHit;
+    case CacheEntryStatus::Unusable:
+      return LookupStatus::CacheMiss;
+    case CacheEntryStatus::RequiresValidation:
+      // The CacheFilter sent the response upstream for validation; check the
+      // filter state to see whether and how the upstream responded. The
+      // filter currently won't send the stale entry if it can't reach the
+      // upstream or if the upstream responds with a 5xx, so don't include
+      // special handling for those cases.
+      switch (filter_state_) {
+      case FilterState::ValidatingCachedResponse:
+        return LookupStatus::RequestIncomplete;
+      case FilterState::EncodeServingFromCache:
+      case FilterState::ResponseServedFromCache:
+        return LookupStatus::StaleHitWithSuccessfulValidation;
+      case FilterState::NotServingFromCache:
+        return LookupStatus::StaleHitWithFailedValidation;
+      default:
```

Comment:
drop `default` and put the IS_ENVOY_BUG after the switch, so the compiler can tell you if you missed an enum choice.